spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,503 @@
+[sample deployment template for a VMSS refering to an existing vnet](https://github.com/Azure/azure-quickstart-templates/blob/master/201-vmss-existing-vnet/azuredeploy.json#L136).
+
+```json
+"properties": {
+  "subnet": {
+    "id": "[resourceId(parameters('existingVnetResourceGroupName'), 'Microsoft.Network/virtualNetworks/subnets', parameters('existingVnetName'), parameters('existingSubNetName'))]"
+  },
+}
+```
+
+#### What ASO does
+Similar to what ARM templates do, ASO uses the decomposition of fully qualified resource id to reference another resource, as seen [here for VMSS -> VNet](https://github.com/Azure/azure-service-operator/blob/92240406aff3863f3a267d8a1dc1e28aa3e841ae/api/v1alpha1/azurevmscaleset_types.go#L25)
+
+```go
+type AzureVMScaleSetSpec struct {
+    ...
+	Location               string `json:"location"`
+	ResourceGroup          string `json:"resourceGroup"`
+	VirtualNetworkName     string `json:"virtualNetworkName"`
+	SubnetName             string `json:"subnetName"`
+    ...
+}
+```
+
+These properties are combined into a fully qualified ARM ID like so:
+
+```go
+subnetIDInput := helpers.MakeResourceID(
+    client.SubscriptionID,
+    resourceGroupName,
+    "Microsoft.Network",
+    "virtualNetworks",
+    vnetName,
+    "subnets",
+    subnetName,
+)
+```
+
+Currently ASO does not support cross-subscription references (and some of the resources such as VMSS above don't even support cross-resource group references), but it in theory could by adding some more parameters.
+
+#### What k8s-infra does
+k8s-infra is a bit different in that resource references are in Kubernetes language (namespace + name) and not Azure language (resource group + resource name).
+All resource references are done using a special type `KnownTypeReference` which contains the fully qualified Kubernetes name for the resource.
+
+### Dependent Resources
+#### What ARM does
+ARM template deployments support [two different ways of deploying dependent resources](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/child-resource-name-type):
+
+- Embed the dependent resource directly in the same ARM template in which the owning resource is created, using the `resources` property.
+    - This requires that the owning resource and the dependent resource are created at the same time.
+- Deploy the dependent resource separately.
+    - When doing this, you should specify `dependsOn` stating that your new dependent resource depends on its owner. This is not strictly required in cases where
+      the owner already exists, but is required when it is still being created (possibly in parallel).
+    - The `name` field of the dependent resource in this mode serves a dual purpose - to name the dependent resource and to identify the parent resources.
+      Each segment of the name corresponds to an owning resource, so creating a Batch Pool `foo` in Batch account `account` would have `name` = `account/foo`.
```

Comment:
The two approaches (embed-directly vs deploy-separately) are described using completely different styles. The former is declarative, the later is first person. Parallel branches (as seen here in bullet lists) should follow similar construction to highlight the similarities/differences between the alternatives.